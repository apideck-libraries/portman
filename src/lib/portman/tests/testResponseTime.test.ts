import { OasMappedOperation, PostmanMappedOperation } from '../../'
import { getOasMappedOperation } from '../../../../__tests__/getOasMappedOperation'
import { getPostmanMappedOperation } from '../../../../__tests__/getPostmanMappedOperation'
import { testResponseTime } from './testResponseTime'

describe('testResponseTime', () => {
  let oasOperation: OasMappedOperation
  let pmOperation: PostmanMappedOperation

  beforeEach(async () => {
    oasOperation = await getOasMappedOperation()
    pmOperation = await getPostmanMappedOperation()
  })

  it('should add test for response time', async () => {
    pmOperation = testResponseTime({ enabled: true, maxMs: 300 }, pmOperation, oasOperation)
    const pmTest = pmOperation.getTests()
    expect(pmTest.script.exec).toMatchSnapshot()
  })
})
