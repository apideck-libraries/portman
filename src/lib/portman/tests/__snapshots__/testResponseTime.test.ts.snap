// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testResponseTime should add test for response time 1`] = `
Array [
  "// Validate response time 
pm.test(\\"[GET]::/crm/companies/:id - Response time is less than 300ms\\", function () {
    pm.expect(pm.response.responseTime).to.be.below(300);
});
",
]
`;
